### PR TITLE
[build] Add comprehensive firmware size reporting scripts for ARM MCUs.

### PR DESCRIPTION
**Pull request by mpaperno**
Adds size reporting utility scripts and changes `firmware-size` target for ARM MCUs to use new report format (on Windows reverts to `arm-none-eabi-size -A` if `bash` not found).

Also in `commit-tests.sh` added firmware size reporting and an option to bypass -Werror.  Fixed up PCBREV option for SKY9X.

Size report examples:
```
---- Size report for firmware.elf (MCU: STM32F205xE) ----

Section Details:

section              size(B)         addr   loc
.text                 220084   0x08000000   Flash
.init_array               32   0x08035bb4   Flash
.fini_array                4   0x08035bd4   Flash
.data (added)           1112   0x08035bd8   Flash
.data                   1112   0x20000000   SRAM1
.bss                   37800   0x20000460   SRAM1
._user_heap_stack       1024   0x20009808   SRAM1

Totals         KB    usage  of ttl	KB free
  Flash:   216.05   42.20%     512	 295.95
  SRAM1:    39.00   34.82%     112	  73.00
  SRAM2:     0.00    0.00%      16	  16.00
```
```
---- Size report for firmware.elf (MCU: ATSAM3S4C) ----

Section Details:
section              size(B)         addr   loc
.text                 197156   0x00400000   Flash
.data (added)           1112   0x00430224   Flash
.data                   1112   0x20000000   SRAM1
.bss                   35624   0x20000460   SRAM1

Totals         KB    usage  of ttl	KB free
  Flash:   193.62   75.63%     256	  62.38
  SRAM1:    35.88   74.74%      48	  12.12
```
```
---- Size report for firmware.elf (MCU: STM32F429xI) ----

Section Details:
section              size(B)         addr   loc
.text                1310248   0x08000000   Flash
.ARM.extab               324   0x0813fe28   Flash
.ARM                     488   0x0813ff6c   Flash
.init_array              108   0x08140154   Flash
.fini_array                4   0x081401c0   Flash
.data (added)           2604   0x081401c4   Flash
.data                   2604   0x10000000   CCRAM
.bss                   40652   0x10000a30   CCRAM
.ram                   12476   0x20000000   SRAM1
._user_heap_stack       1024   0x200030bc   SRAM1
.sdram                783360   0xd0000000   SDRAM2

Totals         KB    usage  of ttl	KB free
  Flash:  1282.98   62.65%    2048	 765.02
  CCRAM:    42.24   66.00%      64	  21.76
  SRAM1:    13.18   11.77%     112	  98.82
  SRAM2:     0.00    0.00%      16	  16.00
  SRAM3:     0.00    0.00%      64	  64.00
 SDRAM2:   765.00    9.34%    8192	7427.00
```
Comparison reports:
```
---- Comparing results of 2 files ----

file 1: firmware-horus-O2.elf
file 2: firmware-horus-Os.elf

Section    file 1   file 2 (Δ vs f1) 
  Flash:  1285.55  1284.83 (  -0.72) 
  CCRAM:    41.64    46.55 (  +4.91) 
  SRAM1:    17.18    17.18 (  --.--) 
  SRAM2:     0.00     0.00 (  --.--) 
  SRAM3:     0.00     0.00 (  --.--) 
 SDRAM2:   765.00   765.00 (  --.--) 
```
```
---- Comparing results of 3 files ----

file 1: firmware1.elf
file 2: firmware2.elf
file 3: firmware3.elf

Section    file 1   file 2 (Δ vs f1)   file 3 (Δ vs f1) (Δ vs f2) 
  Flash:  1285.55  1284.83 (  -0.72)  1286.18 (  +0.63) (  +1.35) 
  CCRAM:    41.64    46.55 (  +4.91)    42.20 (  +0.56) (  -4.35) 
  SRAM1:    17.18    17.18 (  --.--)    17.18 (  --.--) (  --.--) 
```

### CODE REVIEW

**Comment by projectkk2glider** [comment]
Nice one. It works on Linux.

My only concern here is the maintainability of the sh and awk scripts. Doing that stuff in Python would be more to my liking.

**Comment by mpaperno** [comment]
Works in Windows also, with cygwin/msys shell for example.  I'm mostly a Windows user... but seems M$ forgot to include some useful utilities...  :-)

I've used some version of the awk scripts for a couple of years in other projects.  I did think about re-writing it all in Python for OTx, but that is out of scope for now, sorry.  I'm not sure why awk/bash is less "maintainable" though.

It would also be nice if it could parse an ld script and do all the setup automatically... but hey.

File: radio/src/targets/horus/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,9 @@ set(FLAVOUR horus)
 set(RAMBACKUP YES)
 set(LUA YES)
 set(PPM_LIMITS_SYMETRICAL YES)
+# for size report script
+set(CPU_TYPE_FULL STM32F429xI)
```
Comment (mpaperno) [inline radio/src/targets/horus/CMakeLists.txt]: It's a parameter/shortcut for the bash script to set the right memory segment sizes:  https://github.com/opentx/opentx/pull/4218/files#diff-8d137f69d1aafa5d21399516e3d9e0f6R76  
(The pin count doesn't matter and I didn't know it, hence the "x".)

Comment (mpaperno) [inline radio/src/targets/horus/CMakeLists.txt]: I think the next version will parse a .ld script MEMORY section and determine the segments automatically... but I can't promise when that will be done.

Comment (bsongis) [inline radio/src/targets/horus/CMakeLists.txt]: sorry I didn't see we were in HORUS target CMakeLists.txt. No issue here

**Comment by bsongis** [comment]
would need a test on @kilrah windows VM!

**Comment by kilrah** [comment]
Works great on my PC and VM with no config change!

**Comment by bsongis** [comment]
then, go!

**Pull request merged**